Description: open links in browser
 .
 handylinux-thunar-actions (0.3-5) unstable; urgency=medium
 .
   * forgot icon for md5sum check
Author: team handylinux <dev@handylinux.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- handylinux-thunar-actions-0.3.orig/scripts/pixup
+++ handylinux-thunar-actions-0.3/scripts/pixup
@@ -7,11 +7,16 @@ IFS="
 # trads
 IMGSENT="Image sent"
 ERROR="Error. Check file name"
+INTEGCODE="Copy this code to put the picture on the forum"
+IMGADD="Image URL"
+
 
 case $LANG in
 	fr*)
         IMGSENT="Image envoyée"
         ERROR="Erreur à l'envoi d'image. Vérifiez le nom du fichier"
+        INTEGCODE="Copier ce code pour insérer l'image dans le forum"
+        IMGADD="URL de l'image"
         ;;
     de*)
         IMGSENT = "Bild geschickt"
@@ -23,25 +28,48 @@ case $LANG in
         ;;
 esac
 
+
+genhtml() {
+    # create an html page
+    TEMP=$(mktemp pixupXXX.html -p "/tmp")
+    echo "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">" >> $TEMP 
+    echo '<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr" lang="fr">' >> $TEMP
+    echo '<head>' >> $TEMP
+    echo '<meta charset="UTF-8">' >> $TEMP
+    echo '</head>' >> $TEMP
+    echo '<body>' >> $TEMP
+    
+    echo '<style type="text/css">' >> $TEMP
+    echo 'html, body {height: 100%;}' >> $TEMP
+    echo 'html {display: table;margin: auto;}' >> $TEMP
+    echo 'body {display: table-cell;vertical-align: middle; color:"#999"; background-color: #111; }' >> $TEMP
+    echo 'main { padding:1em; border:1px solid #666; 
+    color:#ddd; text-align:center;
+    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.8));}' >> $TEMP
+    echo 'img {border: 4px solid #eee; text-align:center;max-width:80%; margin-left:auto; margin-right:auto}' >> $TEMP
+        
+    echo '</style>' >> $TEMP
+    echo '<main>' >> $TEMP
+    echo "<img src=\"$1\">" >> $TEMP
+    echo "<p>$INTEGCODE :<input onfocus=\"this.select()\" class=\"text\" type=\"text\" value=\"[url=$1][img]$1[/img][/url]\" /></p>" >> $TEMP
+    echo "<p>$IMGADD : <input onfocus=\"this.select()\" class=\"text\" type=\"text\" value=\"$1\" /></p>" >> $TEMP
+
+    echo '</main></body></html>' >> $TEMP
+
+    echo "$TEMP"
+}
+
 for i in $@; do
     TYPE="$(file --mime-type -b $i)"
     EXT="${i##*.}"
-
-    # premier essai sur toile-libre
-    URL=$(curl -L --progress-bar -F "MAX_FILE_SIZE=15360000" \
-        -F "img=@$i;type=$TYPE;filename=$(basename \"$i\")" \
-        "http://pix.toile-libre.org/?action=upload" |grep -E -o "http://pix.toile-libre.org/\?img=[^<]*.$EXT" |head -n1)
-
-    if [ $? -ne 0  -o -z "$URL" ]; then
-        #echec avec toile-libre, on essaie framapic
-        URL=$(curl -F "file=@$i" -F "format=$TYPE"\
-            -F "delete-day=0" -F "first-view=0"\
-            "https://framapic.org" | grep -Eo "https://framapic.org/(.*?).$EXT" | head -n1 | cut -d'"' -f1)
-    fi
+    URL=$(curl -F "file=@$i" -F "format=$TYPE"\
+        -F "delete-day=0" -F "first-view=0"\
+        "https://framapic.org" | grep -Eo "https://framapic.org/(.*?).$EXT" | head -n1 | cut -d'"' -f1)
 
     if [ $? -eq 0 -a -n "$URL" ]; then
+        LOCALHTML=$(genhtml "$URL")
         notify-send "$IMGSENT : $i" &
-        x-www-browser "$URL" &
+        x-www-browser "$LOCALHTML" &
     else
         notify-send "$ERROR" &
     fi
